include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2967,7 +2970,44 @@ class binary_reader
 
     /// the SAX parser
     json_sax_t* sax = nullptr;
+
+    // excluded markers in bjdata optimized type
+#define JSON_BINARY_READER_MAKE_BJD_OPTIMIZED_TYPE_MARKERS_ \
+    make_array<char_int_type>('F', 'H', 'N', 'S', 'T', 'Z', '[', '{')
+
+#define JSON_BINARY_READER_MAKE_BJD_TYPES_MAP_ \
+    make_array<bjd_type>(                      \
+    bjd_type{'C', "char"},                 \
+    bjd_type{'D', "double"},               \
+    bjd_type{'I', "int16"},                \
+    bjd_type{'L', "int64"},                \
+    bjd_type{'M', "uint64"},               \
+    bjd_type{'U', "uint8"},                \
+    bjd_type{'d', "single"},               \
+    bjd_type{'i', "int8"},                 \
+    bjd_type{'l', "int32"},                \
+    bjd_type{'m', "uint32"},               \
+    bjd_type{'u', "uint16"})
+
+  JSON_PRIVATE_UNLESS_TESTED:
+    // lookup tables
+    // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+    const decltype(JSON_BINARY_READER_MAKE_BJD_OPTIMIZED_TYPE_MARKERS_) bjd_optimized_type_markers =
```

Comment:
Can this be `static const` and then you can use `auto` and not have to have the `decltype` and macro to repeat?

Reply:
I don't think so … that'll just end in clang-tidy (or is it a Clang warning flag?) complaining.

Reply:
What would it be complaining about?

Reply:
Exit-time destructors, same as last time we had this conversation. :-)
Don't remember which PR...

Reply:
Ah, thanks, didn't remember that.